lib/manifest.ts

```diff
@@ -250,6 +247,30 @@ export class Manifest {
     );
   }
 
+  /**
+   * Validates that `assumeRoleAdditionalOptions` doesn't contain nor `ExternalId` neither `RoleArn`, as they
+   * have dedicated properties preceding this (e.g `assumeRoleArn` and `assumeRoleExternalId`).
+   */
+  private static validateAssumeRoleAdditionalOptions(
+    instance: any,
+    _key: string,
+    _schema: jsonschema.Schema,
+    _options: jsonschema.Options,
+    _ctx: jsonschema.SchemaContext
+  ) {
+    if (_key !== 'assumeRoleAdditionalOptions') {
+      return;
+    }
+
+    const assumeRoleOptions = instance.assumeRoleAdditionalOptions;
```

Comment:
Been going back and forth on whether this belongs in the CLI or here. 

- **Argument for CLI**: The CLI is the deployment tool that actually suffers from this duplication, why are we preventing other deployment tools from creating such manifests?
- **Argument for here**: Creating a manifest with both `assumeRoleArn` and `assumeRoleAdditionalOptions.RoleArn` is confusing, and we should disallow it altogether, regardless of which deployment tool will act on the schema. 

I think its unlikely other deployment tools will choose to ignore this duplication, so I put it here. It also has the benefit of not duplicating the validation on both the CLI and cdk-assets. 

Interested in thoughts here though.

Reply:
I think it makes sense to consider this part of the schema.

We could/would express this as part of the JSON Schema if we could, so treating it equivalently seems solid to me.